src/Illuminate/Collections/Arr.php

```diff
@@ -708,15 +695,30 @@ public static function set(&$array, $key, $value)
      */
     public static function shuffle($array, $seed = null)
     {
-        if (is_null($seed)) {
-            shuffle($array);
-        } else {
+        if (! is_null($seed)) {
             mt_srand($seed);
             shuffle($array);
             mt_srand();
+
+            return $array;
         }
 
-        return $array;
+        $keys = array_keys($array);
+
+        for ($i = count($keys) - 1; $i > 0; $i--) {
+            $j = random_int(0, $i);
+            $temp = $keys[$i];
+            $keys[$i] = $keys[$j];
+            $keys[$j] = $temp;
+        }
+
+        $shuffled = [];
+
+        foreach ($keys as $key) {
+            $shuffled[$key] = $array[$key];
```

Comment:
I wrote the code from scratch on my end, and got the same result, except that in this line I had:

`$shuffled[] = $array[$key];`

Indeed, [shuffle()](https://www.php.net/manual/en/function.shuffle.php) does not preserve the keys, not even the string keys.

Reply:
As the keys should not be preserved, we could use `array_values()`, so that `$shuffled` can be made in one iteration, instead of needing a second iteration to build it from the shuffled keys:

```php
$shuffled = array_values($array);
for ($i = count($shuffled) - 1; $i >= 1; $i--) {
    $j = random_int(0, $i);
    $temp = $shuffled[$i];
    $shuffled[$i] = $shuffled[$j];
    $shuffled[$j] = $temp;
}
return $shuffled;
```

It is shorter, and a bit faster.